### PR TITLE
Update install instructions for snaps

### PR DESCRIPTION
**Pull request by popey**
This pull request adds installation instructions for users of [snap enabled Linux distributions](https://snapcraft.io/docs/core/install). Snaps enable ISVs and projects to make their software available to millions of Linux users via the snap store and directly control delivery of software updates to their users. Snaps of stable desktop applications will also be available via the desktop Software Center, improving the discover-ability of your software. Your software can be installed with a simple `snap install mattermost-desktop --beta` simplifying the installation instructions for your users.

Once the snap is released into the stable channel in the store, users no longer need to add the ```--beta``` and will appear in the desktop Software Center

### CODE REVIEW

**Comment by JeffSchering** [comment]
For context, see [[Help Wanted] - Snaps package for Mattermost #4554](https://github.com/mattermost/platform/issues/4554)

**Comment by jasonblais** [comment]
This is great

@JeffSchering what are your thoughts on re-formatting the section slightly, so it's clear there are three ways to deploy the desktop application in the Linux environments (tar.gz, debian package or snaps package)

**Comment by popey** [comment]
Apologies for overlooking the CLA requirement. I've signed this now.

**Comment by JeffSchering** [comment]
@jasonblais I think reformatting the section is essential... it'll be part of my review, which I hope to get to later today.

File: source/install/desktop.rst [hunk 1]
```diff
@@ -59,6 +59,10 @@ Linux (Beta)
 
 3. If you need a Desktop Entry for the application to integrate into a desktop environment, please refer to https://wiki.archlinux.org/index.php/Desktop_entries
 
+To install the snap on [snap supported](https://snapcraft.io/docs/core/install) systems from the store.
+
+    snap install mattermost-desktop --beta
```
Comment (JeffSchering) [inline source/install/desktop.rst]: Do you need sudo for this? The example on https://snapcraft.io/ uses sudo and I was just wondering if it's required for the desktop app.

**Comment by JeffSchering** [comment]
@popey How do you start Mattermost after installing it with snapcraft?

**Comment by popey** [comment]
@JeffSchering You can either launch it from the command line ```mattermost-desktop``` or the traditional desktop way by clicking the Mattermost icon in the desktop launcher/menu.

**Comment by JeffSchering** [comment]
@popey Thanks! We've already had some positive feedback about this on our forum.

I updated the instructions that you provided to fit in better with the doc, but also did some rewording. Can you confirm that it's correct? I tried to create a PR instead of committing directly to your branch, but GitHub threw a 404 when I tried that.

**Comment by popey** [comment]
Hey @JeffSchering the section "A snap is available for systems that have Snapcraft installed. Snapcraft is installed by default on Ubuntu 16.04 and later, but for most other Linux distributions you can install it manually. To install Snapcraft," needs to have s/snapcraft/snapd/. Snapcraft is the tool to build snaps, snapd is the daemon which enables snaps to be installed. Sudo is only necessary if you haven't done "snap login". Once you do that, sudo is no longer required. If you don't mention it in the intstructions that's fine, because "snap install" will remind you to use sudo if it needs it.

File: source/install/desktop.rst [hunk 2]
```diff
@@ -50,25 +50,52 @@ The Mattermost desktop should open automatically.
...
+    64-bit systems
+      ``sudo dpkg -i mattermost-desktop-3.6.0-linux-amd64.deb``
+    32-bit systems
+      ``sudo dpkg -i mattermost-desktop-3.6.0-linux-i386.deb``
 
-3. If you need a Desktop Entry for the application to integrate into a desktop environment, please refer to https://wiki.archlinux.org/index.php/Desktop_entries
+3. To run Mattermost, open **Dash** (located at top left corner) and input ``mattermost``, then click the Mattermost icon.
 
-To download the unofficial, community-driven .deb packages for the desktop application:
+Generic Linux package
```
Comment (jasonblais) [inline source/install/desktop.rst]: Can we move the generic Linux package first, since the Debian packages are unofficial?

**Comment by jasonblais** [comment]
@yuya-oc would you like to help with review as well?

**Comment by jasonblais** [comment]
And a huge thanks @popey for this, really excited for the snaps package

File: source/install/desktop.rst (hunk 2 above)
Comment (JeffSchering) [inline source/install/desktop.rst]: Done.

**Pull request merged**